cmem/encoder_test.go

```diff
@@ -6,51 +6,50 @@ package cmem
 
 import (
 	"bytes"
-	"encoding/binary"
-	"strconv"
 	"testing"
+	"encoding/binary"
 )
 
 func TestEncode(t *testing.T) {
-	for i, tc := range []struct {
-		v    interface{}
-		want []byte
-	}{
-		{
-			v: struct {
-				V1 uint8
-				V2 uint64
-				V3 uint8
-				V4 uint16
-			}{1, 2, 3, 4},
-			want: []byte{
-				0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // |........|
-				0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // |........|
-				0x03, 0x00, 0x04, 0x00, /*                   */ // |....|
-			},
-		},
-		{
-			v: []int32{1, 20, 300, 4000, 50000, 600000, 7000000, 8000000},
-			want: []byte{
-				0x01, 0x00, 0x00, 0x00, 0x14, 0x00, 0x00, 0x00, // |........|
-				0x2c, 0x01, 0x00, 0x00, 0xa0, 0x0f, 0x00, 0x00, // |,.......|
-				0x50, 0xc3, 0x00, 0x00, 0xc0, 0x27, 0x09, 0x00, // |P....'..|
-				0xc0, 0xcf, 0x6a, 0x00, 0x00, 0x12, 0x7a, 0x00, // |..j...z.|
-			},
-		},
-	} {
-		t.Run(strconv.Itoa(i), func(t *testing.T) {
-			oldEndian := nativeEndian
-			nativeEndian = binary.LittleEndian
-			defer func() { nativeEndian = oldEndian }()
-			var enc Encoder
-			err := enc.Encode(tc.v)
-			if err != nil {
-				t.Fatalf("could not encode: %v", err)
-			}
-			if !bytes.Equal(enc.Buf, tc.want) {
-				t.Fatalf("encoding error:\ngot = %v\nwant= %v", enc.Buf, tc.want)
-			}
-		})
+	v := struct {
+		V1 uint8
+		V2 uint64
+		V3 uint8
+		V4 uint16
+	}{1, 2, 3, 4}
+	want := []byte{
+		0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // |........|
+		0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // |........|
+		0x03, 0x00, 0x04, 0x00, /*                   */ // |....|
+	}
+	var enc Encoder
+	err := enc.Encode(v)
+	if err != nil {
+		t.Fatalf("could not encode: %v", err)
 	}
+	if !bytes.Equal(enc.Buf, want) {
+		t.Fatalf("encoding error:\ngot = %v\nwant= %v", enc.Buf, want)
+	}
+}
+
+func TestEncodeSlice(t *testing.T) {
+	v := []int32{1, 20, 300, 4000, 50000, 600000, 7000000, 8000000}
+	want := []byte{
+		0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // |........|
```

Comment:
This was manually created and is not correct.

Reply:
Based on the HDF5 hvl_t strcture, 
/* Variable Length Datatype struct in memory */
/* (This is only used for VL sequences, not VL strings, which are stored in char *'s) */
typedef struct {
    size_t len; /* Length of VL data (in base type units) */
    void *p;    /* Pointer to VL data */
} hvl_t;

The variable length array in memory has two fields: len and pointer to the heap memory which is holding the actual data.

In this test case, the first 8 bytes indicates the length of the data: 8